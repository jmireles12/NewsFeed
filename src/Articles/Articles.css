.Article {
    padding-bottom: 10px;
    text-align: center;
}

a {
    color: black;
    text-decoration: none;
}

a:hover {
    color: #495057;
}

.photo {
    height: 200px;
    width: 300px;
    padding-top: 10px;
}

.title {
    font-size: medium;
}