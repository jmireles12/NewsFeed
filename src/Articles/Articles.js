import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import ApiContext from '../ApiContext'
import './Articles.css'

export default class Articles extends Component {

    static defaultProps = {
        match: {
            params: {}
        }
    }

    static contextType = ApiContext;

    render() {
        const { url, title, thumbnail } = this.props
        return (
            <section className='Article'>
                <img className='photo' src={thumbnail} alt='provided by Guardian.com'/>
                <h2 className='title'>
                    <Link to={{ pathname: url }} target='_blank'>{title}</Link>
                </h2>
            </section>
        )
    }
}