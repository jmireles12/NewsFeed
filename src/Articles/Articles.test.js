import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Articles from './Articles'

describe(`Articles component`, () => {
    it('renders the complete Articles', () => {
        const wrapper = shallow(<Articles />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})