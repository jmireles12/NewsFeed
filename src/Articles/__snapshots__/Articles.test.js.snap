// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles component renders the complete Articles 1`] = `
<section
  className="Article"
>
  <img
    alt="provided by Guardian.com"
    className="photo"
  />
  <h2
    className="title"
  >
    <Link
      target="_blank"
      to={
        Object {
          "pathname": undefined,
        }
      }
    />
  </h2>
</section>
`;
